### PR TITLE
Group page and other edits

### PR DESCRIPTION
**Pull request by philjng**
Homepage color scheme added
image

**user state updated to:**
```
const initialState = {
    isAuth: false,
    user_id: 0,
    userEvents: [],
    userGroups: {
        owned: [],
        joined: []
    }
};
```

**New group page**
- can add / remove groups
- will persist to homepage
image

### CODE REVIEW

File: src/reducers/user.js [hunk 1]
```diff
@@ -1,29 +1,55 @@
...
+            return {
+                ...state,
+                isAuth: false,
+                user_id: null,
+            };
+        case "ADD_GROUP": {
+            return {
+                ...state,
+                userGroups: {
+                    ...state.userGroups.owned,
```
Comment (tsuiswz) [inline src/reducers/user.js:34]: isnt it better to just use `...state.userGroups,`?

File: src/components/Groups/Groups.js [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+
+const AllGroups = styled(Box1)({
+    padding: `1rem`,
+    minWidth: `50vw`,
+    backgroundColor: `#ebfaf7`
+})
+
+const GroupsContainer = (props) => {
+    const filtered = props.groups.filter(
+        (group) => !props.userGroups.joined.map((group) => group.groupId).includes(group.groupId))
```
Comment (tsuiswz) [inline src/components/Groups/Groups.js:24]: Would it be better to take out the !props.userGroups.joined.map((group) => group.groupId) portion as a const above so like:

const userGroupIds = props.userGroups.joined.map((userGroup) => userGroup.groupId)
const filtered = props.groups.filter((group) => !userGroupIds.includes(group.groupId))

It's kinda of convoluted to have it all on one line and hard to read. The filter and map function passes in a parameter that are named the same, so its even harder to dissect. Also you call the map function on the same props.userGroups every time you iterate through with the filter function, so you can pull it out for inconsequential performance reasons.

File: src/reducers/user.js (hunk 1 above)
Comment (philjng) [reply src/reducers/user.js:34]: what do you mean?

**Comment by philjng** [comment]
image

**Pull request merged**